Jenkinsfile

```diff
@@ -5,6 +5,9 @@ pipeline {
   agent any
   stages {
     stage('Execute CI') {
+      options {
+        timeout(time: 3, unit: 'HOURS')
```

Comment:
3 hours seems like a safe choice for now, given that the nominal time seems to be ~35 minutes recently. We could potentially set this much lower, but my thinking is that we mainly want this to prevent zombie runs from clogging the pipeline, so a longer timeout should be adequate.

Reply:
now I want to know what are all the units...